### PR TITLE
Introduce `-version` option to display version

### PR DESCRIPTION
**Pull request by mingjun97**
## Description

* Generate a file that holds git hash and version at build time.
* Add option that can help print out the version info. This could benifit manage the versioning stuffs in `Issues`.

## Type of changes

- [ ] Bug fix
- [x] New feature 
- [ ] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

**Comment by mingjun97** [comment]
Well.. Test locally and it works fine. But github action reject the changes. I'll play with it to figure out a solution.

**Comment by mcxiaoke** [comment]
also add this to server start log message better , and append to the output jar name

File: src/main/java/emu/grasscutter/Grasscutter.java [hunk 1]
```diff
@@ -79,6 +80,9 @@ public static void main(String[] args) throws Exception {
 				case "-gachamap" -> {
 					Tools.createGachaMapping(Grasscutter.getConfig().DATA_FOLDER + "/gacha_mappings.js"); exitEarly = true;
 				}
+				case "-version" -> {
```
Comment (4Benj) [inline src/main/java/emu/grasscutter/Grasscutter.java:83]: Why do we need a version argument?

File: src/main/resources/languages/en-US.json [hunk 2]
```diff
@@ -45,7 +45,8 @@
       "run_mode_error": "Invalid server run mode: %s.",
       "run_mode_help": "Server run mode must be 'HYBRID', 'DISPATCH_ONLY', or 'GAME_ONLY'. Unable to start Grasscutter...",
       "create_resources": "Creating resources folder...",
-      "resources_error": "Place a copy of 'BinOutput' and 'ExcelBinOutput' in the resources folder."
+      "resources_error": "Place a copy of 'BinOutput' and 'ExcelBinOutput' in the resources folder.",
+      "version": "Grasscutter version: %s, Git Hash: %s"
```
Comment (4Benj) [inline src/main/resources/languages/en-US.json]: Recommended layout 1.1.0-1234a88
Don’t seperate the version and hash

File: build.gradle [hunk 3]
```diff
@@ -113,6 +122,13 @@ jar {
     from('src/main/java') {
         include '*.xml'
     }
+    new File(projectDir, "src/main/java/emu/grasscutter/BuildConfig.java").text = """
```
Comment (4Benj) [inline build.gradle]: Maybe something like [this](https://stackoverflow.com/a/33026919) will be better?

**Review by 4Benj — CHANGES_REQUESTED** [review]

File: src/main/java/emu/grasscutter/Grasscutter.java (hunk 1 above)
Comment (mingjun97) [reply src/main/java/emu/grasscutter/Grasscutter.java:83]: Magix mentioned we probably want to ask user put the version inside the issue. A version argument should be relatively easy for users to retrieve the version information.

Comment (4Benj) [reply src/main/java/emu/grasscutter/Grasscutter.java:83]: I feel like the version should be printed/logged on start up instead. If it already is, then I think this is unneeded redundancy.

File: build.gradle (hunk 3 above)
Comment (mingjun97) [reply build.gradle]: I'll try to implement it in this way.

File: src/main/java/emu/grasscutter/Grasscutter.java (hunk 1 above)
Comment (mingjun97) [reply src/main/java/emu/grasscutter/Grasscutter.java:83]: I also implemented printed the version out at this commit. [5e74897](https://github.com/Grasscutters/Grasscutter/pull/797/commits/5e74897b994435b20e788b46507eaacaa2917831)

I can remove the `-version` argument if you like. I still think it's useful in some cases though.

File: build.gradle (hunk 3 above)
Comment (mingjun97) [reply build.gradle]: Fixed in the [5aefad6](https://github.com/Grasscutters/Grasscutter/pull/797/commits/5aefad61eacfad17c46a4f139b5d5c8f56c0aff7).

**Review by 4Benj — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**